### PR TITLE
curvefs/metaserver: fix trash bugs

### PR DESCRIPTION
**Pull request by wuhongsong**
### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by wuhongsong** [comment]
cicheck

**Comment by wuhongsong** [comment]
cicheck

**Comment by wuhongsong** [comment]
cicheck

**Comment by wuhongsong** [comment]
cicheck

**Comment by wuhongsong** [comment]
cicheck

**Comment by wuhongsong** [comment]
cicheck

File: curvefs/src/metaserver/inode_manager.cpp [hunk 1]
```diff
@@ -312,7 +312,7 @@ MetaStatusCode InodeManager::DeleteInode(uint32_t fsId, uint64_t inodeId,
 MetaStatusCode InodeManager::UpdateInode(const UpdateInodeRequest& request,
                                          int64_t logIndex) {
     CHECK_APPLIED();
-    VLOG(9) << "update inode, fsid: " << request.fsid()
+    VLOG(0) << "update inode, fsid: " << request.fsid()
```
Comment (SeanHai) [inline curvefs/src/metaserver/inode_manager.cpp:315]: VLOG(0) is too often

File: curvefs/src/metaserver/inode_manager.cpp [hunk 2]
```diff
@@ -443,7 +438,14 @@ MetaStatusCode InodeManager::UpdateInode(const UpdateInodeRequest& request,
...
     }
-    VLOG(9) << "UpdateInode success, " << request.ShortDebugString();
+
+    if (s3NeedTrash) {
+        inodeStorage_->UpdateDeletingKey(old, logIndex);
+        trash_->Add(old.fsid(), old.inodeid(), old.dtime());
+        --(*type2InodeNum_)[old.type()];
+    }
+
+    VLOG(0) << "UpdateInode success, " << request.ShortDebugString();
```
Comment (SeanHai) [inline curvefs/src/metaserver/inode_manager.cpp:448]: Ditto

File: curvefs/src/metaserver/inode_storage.h [hunk 3]
```diff
@@ -187,6 +198,11 @@ class InodeStorage {
     MetaStatusCode GetAllBlockGroup(
         std::vector<DeallocatableBlockGroup>* deallocatableBlockGroupVec);
 
+/*
+    std::shared_ptr<KVStorage> GetKVStorage() {
```
Comment (SeanHai) [inline curvefs/src/metaserver/inode_storage.h:202]: remove it if needn't

File: curvefs/src/metaserver/storage/rocksdb_storage.h [hunk 4]
```diff
@@ -92,6 +95,13 @@ class RocksDBStorage : public KVStorage, public StorageTransaction {
                 const std::string& key,
                 const ValueType& value) override;
 
+    Status HSetDeleting(const std::string& name,
```
Comment (SeanHai) [inline curvefs/src/metaserver/storage/rocksdb_storage.h:98]: The interfaces added is unnecessary, rocksdb_storage only need the most basic interface, and you should deal the logic upper layer

File: curvefs/src/metaserver/inode_storage.cpp [hunk 5]
```diff
@@ -185,9 +185,74 @@ MetaStatusCode InodeStorage::Insert(const Inode& inode, int64_t logIndex) {
     return MetaStatusCode::STORAGE_INTERNAL_ERROR;
 }
 
+MetaStatusCode InodeStorage::UpdateDeletingKey(
+  const Inode& inode, int64_t logIndex) {
+    WriteLockGuard lg(rwLock_);
+    Key4Inode key(inode.fsid(), inode.inodeid());
```
Comment (SeanHai) [inline curvefs/src/metaserver/inode_storage.cpp:191]: Here you can add a new struct like `Key4DeletedInode` in which you can add deleted prefix before inode key

File: curvefs/src/metaserver/inode_storage.h [hunk 6]
```diff
@@ -43,6 +43,7 @@
 #include "curvefs/src/metaserver/storage/utils.h"
 #include "src/common/concurrent/rw_lock.h"
 
+#define DELETING_PREFIX "deleting_"
```
Comment (SeanHai) [inline curvefs/src/metaserver/inode_storage.h:46]: The raw string prefix is inappropriate, you can add a new key type in https://github.com/opencurve/curve/blob/dd8ce61654940a8e789960ce5c03c1c84b09bf0a/curvefs/src/metaserver/storage/converter.h#L40

File: curvefs/src/metaserver/inode_storage.cpp [hunk 7]
```diff
@@ -185,9 +185,74 @@ MetaStatusCode InodeStorage::Insert(const Inode& inode, int64_t logIndex) {
...
+    std::string skey = conv_.SerializeToString(key);
+    VLOG(9) << "update deleting key, " << inode.inodeid();
+    const char* step = "Begin transaction";
+    std::shared_ptr<storage::StorageTransaction> txn;
+    txn = kvStorage_->BeginTransaction();
+    if (txn == nullptr) {
+        LOG(ERROR) << "Begin transaction failed";
+        return MetaStatusCode::STORAGE_INTERNAL_ERROR;
+    }
+    auto rc = txn->HSetDeleting(table4Inode_, skey , inode);
```
Comment (SeanHai) [inline curvefs/src/metaserver/inode_storage.cpp:201]: The dtime as the value set to rocksdb is enough

**Pull request closed without merging**